openmdao/surrogate_models/kriging.py

```diff
@@ -91,6 +92,12 @@ def _declare_options(self):
                                   "or 'gesvd' which is slower but more reliable."
                                   "'gesvd' is the default.")
 
+        self.options.declare('cache_trained_model', default=False, types=bool,
+                             desc='Cache the trained model to avoid repeat training')
```

Comment:
You could probably get a way with just having one option 'cached_model_filename', that can be set to None when you don't want to load a cache. (I guess what you lose then is the standardized default filename.)

Maybe you could also have a similar option to save the surrogate model to a a pickle.